docs/site/Authentication-component-decorator.md

```diff
@@ -0,0 +1,101 @@
+  constructor(
+    // `AuthenticationBindings.CURRENT_USER` is now an alias of
+    // `SecurityBindings.USER` in @loopback/security
+    // ------ ADD SNIPPET ---------
+    @inject(SecurityBindings.USER)
+    private userProfile: UserProfile,
+    // ------------- END OF SNIPPET -------------
+  )
+  {}
+
+  // ------ ADD SNIPPET ---------
+  @authenticate('basic')
+  @get('/whoami')
+  whoAmI(): string {
+    // `securityId` is Symbol to represent the security id of the user,
+    // typically the user id. You can find more information of `securityId` in
+    // https://loopback.io/doc/en/lb4/Security
+    return this.userProfile[securityId];
+  }
+  // ------------- END OF SNIPPET -------------
+}
+```
+
+{% include note.html content="If only <b>some</b> of the controller methods are decorated with the <b>@authenticate</b> decorator, then the injection decorator for SecurityBindings.USER in the controller's constructor must be specified as <b>@inject(SecurityBindings.USER, {optional:true})</b> to avoid a binding error when an unauthenticated endpoint is accessed. Alternatively, do not inject SecurityBindings.USER in the controller <b>constructor</b>, but in the controller <b>methods</b> which are actually decorated with the <b>@authenticate</b> decorator. See [Method Injection](Dependency-injection.md#method-injection), [Constructor Injection](Dependency-injection.md#constructor-injection) and [Optional Dependencies](Dependency-injection.md#optional-dependencies) for details.
+" %}
+
+An example of the decorator when options **are** specified looks like this:
+
+```ts
+@authenticate('basic', { /* some options for the strategy */})
+```
+
+{% include tip.html content="
+To avoid repeating the same options in the <b>@authenticate</b> decorator for many endpoints in a controller, you can instead define global options which can be injected into an authentication strategy thereby allowing you to avoid specifying the options object in the decorator itself. For controller endpoints that need to override a global option, you can specify it in an options object passed into the decorator. Your authentication strategy would need to handle the option overrides. See [Managing Custom Authentication Strategy Options](Loopback-component-options.md) for details.
+" %}
+
+After a request is successfully authenticated, the current user profile is
+available on the request context. You can obtain it via dependency injection by
+using the `SecurityBindings.USER` binding key.
+
+Parameters of the `@authenticate` decorator can be obtained via dependency
+injection using the `AuthenticationBindings.METADATA` binding key. It returns
+data of type `AuthenticationMetadata` provided by `AuthMetadataProvider`. The
+`AuthenticationStrategyProvider`, discussed in a later section, makes use of
+`AuthenticationMetadata` to figure out what **name** you specified as a
+parameter in the `@authenticate` decorator of a specific controller endpoint.
+
```

Comment:
I feel like this paragraph contains too much information. Is it possible to just provide minimum information to explain the mechanism?

For example ( I might not understand it correctly):
With  the binding key `AuthenticationBindings.METADATA`, the
`AuthenticationStrategyProvider`, which will be discussed in a later section, can figure out what strategy name is specified in the `@authenticate` decorator.

Cause I assume that at this point the read should know how binding key works, and `@authenticate` decorates the endpoint.

Reply:
@agnes512 Not really, people don't always read all docs in sequence...personally I feel this paragraph is good and necessary, so I tend to keep it as is. 
(but thank you for writing the draft 👍 )